src/autowiring/test/AutowiringTest.cpp

```diff
@@ -99,6 +99,8 @@ class DynamicCastDerivedType : public PublicBase, private PrivateBase {
 public:
   DynamicCastDerivedType(void) {}
   ~DynamicCastDerivedType(void) {}
+
+  FRIEND_TEST(AutowiringTest, PostHocTypeAdvertisement);
```

Comment:
Yes but please, please resist the temptation to use this anywhere except in test code.  I hate it when unit tests infiltrate an otherwise pristine codebase.